File: expo-maps/android/src/main/java/expo/modules/maps/interfaces/ExpoMapView.kt
package expo.modules.maps.interfaces

import expo.modules.maps.*

interface ExpoMapView {
  fun setMapType(mapType: MapType)
  fun setMarkers(markerObjects: Array<MarkerObject>)
  fun setPolygons(polygonObjects: Array<PolygonObject>)
  fun setPolylines(polylineObjects: Array<PolylineObject>)
  fun setCircles(circleObjects: Array<CircleObject>)
  fun setClusters(clusterObjects: Array<ClusterObject>)
  fun setEnabledTraffic(enableTraffic: Boolean)
  fun setKMLs(kmlObjects: Array<KMLObject>)
  fun setGeoJsons(geoJsonObjects: Array<GeoJsonObject>)
  fun setInitialCameraPosition(initialCameraPosition: CameraPosition)
  fun setOverlays(overlayObjects: Array<OverlayObject>)
}

File: expo-maps/ios/ExpoMaps/Protocols/ExpoMapView.swift
protocol ExpoMapView: UIView {
  init()
  init?(coder: NSCoder)

  func setMapType(mapType: MapType)
  func setMarkers(markerObjects: [MarkerObject])
  func setPolygons(polygonObjects: [PolygonObject])
  func setPolylines(polylineObjects: [PolylineObject])
  func setInitialCameraPosition(initialCameraPosition: CameraPosition)
  func setEnabledTraffic(enableTraffic: Bool)
  func setKMLs(kmlObjects: [KMLObject])
  func setGeoJsons(geoJsonObjects: [GeoJsonObject])
  func setOverlays(overlayObjects: [OverlayObject])
}
